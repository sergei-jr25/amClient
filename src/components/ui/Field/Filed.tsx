import { forwardRef } from 'react'

import { IField } from './Field.interface'
import styles from './Field.module.scss'

const Filed = forwardRef<HTMLInputElement, IField>(
	({ error, placeholder, type, ...rest }, ref) => {
		return (
			<label className={styles.field}>
				<button></button>
				<input
					ref={ref}
					className={styles.field__input}
					placeholder={placeholder}
					type={type}
					name='input'
					{...rest}
				/>
				<span className={styles.field__error}>{error?.message}</span>
			</label>
		)
	}
)
export default Filed
