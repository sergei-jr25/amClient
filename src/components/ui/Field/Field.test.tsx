import '@testing-library/jest-dom'

import { render, screen } from '@testing-library/react'
import Filed from './Filed'

describe('Field Component', () => {
	it('renders input with correct placeholder and type', () => {
		const placeholder = 'Placeholder'
		const type = 'text'

		render(<Filed placeholder='Placeholder' type={type} />)
		expect(screen.getByPlaceholderText('Placeholder')).toHaveAttribute(
			'type',
			type
		)
	})

	it('renders label with correct text and input with correct placeholder and type', () => {
		const type = 'text'
		render(<Filed placeholder='Placeholder' type={type} />)
		const labelElement = screen.getByPlaceholderText('Placeholder')
		expect(labelElement).toHaveClass('field__input')
		expect(labelElement).toMatchSnapshot()
	})

	it('Chacked error message', () => {
		const errorMessage = 'This is an error message'
		render(<Filed error={{ message: errorMessage }} />)
		const spanError = screen.getByText(errorMessage)
		expect(spanError).toBeInTheDocument()
	})
})
