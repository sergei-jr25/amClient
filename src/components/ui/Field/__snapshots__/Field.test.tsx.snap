// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Component renders label with correct text and input with correct placeholder and type 1`] = `
<input
  class="field__input"
  name="input"
  placeholder="Placeholder"
  type="text"
/>
`;
